linkedlist.c

```diff
@@ -0,0 +1,162 @@
+/*Jinoo Hong*/
+/*Write a program linkedlist.c creating a node structure as described in class. Implement the following functions for integers:
+	printList(): given a linked list, print out the list
+	add(): given a linked list and an integer, add a node with the integer to the beginning of the list.
+	remove(): given a linked list and an integer, remove the first instance of that integer in the list. Print an error message if the integer is not found.
+	clear(): given a linked list, empty the entire linked list (delete all nodes).
+	smallestLength(): Given two linked lists of integers, write a function to find the length of the smallest list.
+	smallestSum():Given two linked lists of integers, write a function to find the list with the smallest sum.*/
+
+#include<stdio.h>
+#include<stdlib.h>
+
+
+struct node {
+	int data;
+	struct node *next;
+};
+
+
+//creating a linked list
+void insert(struct node *pointer, int data) {
+	while(pointer->next!=NULL) {
+		pointer = pointer -> next;
+	}
+	pointer->next = (struct node *)malloc(sizeof(struct node));
+	pointer = pointer->next;
+	pointer->data = data;
+	pointer->next = NULL;
+}
+
+
+//printing the list
+void printList(struct node *pointer) {
+	printf("List: ");
+	pointer = pointer->next;
+	while(pointer->next!=NULL) {
+		printf("%d ",pointer->data);
+		pointer = pointer->next;
+	}
+	printf("\n");
+}
+
+
+//adding to the front of the list
+void addingtofront(struct node *head, int data) {
+	struct node *new;
+	new=(struct node *) malloc (sizeof(struct node));
+	new->next=head->next;
+	head->next=new;
+	new->data=data;
+	printList(head);
+}
+
+//deleting an 18
+void delete(struct node *pointer, int numbertofind) {
+	struct node *head = pointer;
+	int end = 0;
+	while(pointer->next->data!=numbertofind) {
+		pointer=pointer->next;
+		if (pointer->next==NULL){
+			printf("Number not found in list\n");
+			end = 1;
+		}
+	}
+	if (end==0) {
+		struct node *temp;
+		temp = pointer->next;
+		pointer->next=pointer->next->next;
+		free(temp);
+		printList(head);
+	}
+}
+
+//clearing/freeing the list
+void clear (struct node *pointer) {
+	struct node *temp;
+	while (pointer->next!=NULL) {
+		temp = pointer;
+		free(pointer);
+		pointer=temp;
+		pointer=pointer->next;
+	}
+		printf("The list has been cleared\n");
+}
+
+int findsmaller(struct node *pointer1, struct node *pointer2) {
+	int counter;
+	for (counter = 0; pointer1->next!=NULL&&pointer2->next!=NULL;counter++, pointer1=pointer1->next, pointer2=pointer2->next) {
+	}
+	return counter-1;
+}
+
+int findsum(struct node *pointer1, struct node *pointer2) {
```

Comment:
this function returned a warning! the reason is because there isn't an automatic value being returned. also, this should have returned the list that was had a smaller sum instead.